apps/api/src/app/events/usecases/trigger-event/message-matcher.service.spec.ts

```diff
@@ -557,6 +558,189 @@ describe('Message filter matcher', function () {
 
     gotGetStub.restore();
   });
+
+  describe('is online filters', () => {
+    const getSubscriber = (
+      { isOnline }: { isOnline?: boolean } = {},
+      { subDuration }: { subDuration?: Duration } = {}
+    ) => ({
+      firstName: 'John',
+      lastName: 'Doe',
+      isOnline: isOnline ?? true,
+      lastOnlineAt: subDuration ? sub(new Date(), subDuration).toISOString() : new Date().toISOString(),
+    });
+
+    describe('isOnline', () => {
+      it('allows to process if the subscriber is online', async () => {
+        const matcher = new MessageMatcher(
+          { findOne: () => Promise.resolve(getSubscriber()) } as any,
+          undefined as any,
+          undefined as any
+        );
```

Comment:
I couldn't find a better way how to pass a mocked `SubscriberRepository` instance with Sinon to the `MessageMatcher`, so decided to do that this way.